Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -34,75 +37,73 @@
                                     'for GetEntityByID is 30 days.'
 
 
-class Client:
-    def __init__(self, self_deployed: bool, refresh_token: str, auth_and_token_url: str, enc_key: str,
-                 redirect_uri: str, auth_code: str, subscription_id: str, resource_group_name: str, workspace_name: str,
-                 verify: bool, proxy: bool):
-
-        tenant_id = refresh_token if self_deployed else ''
-        refresh_token = get_integration_context().get('current_refresh_token') or refresh_token
-        base_url = f'https://management.azure.com/subscriptions/{subscription_id}/' \
-                   f'resourceGroups/{resource_group_name}/providers/Microsoft.OperationalInsights/workspaces/' \
-                   f'{workspace_name}/providers/Microsoft.SecurityInsights'
-        self.ms_client = MicrosoftClient(
-            self_deployed=self_deployed,
-            auth_id=auth_and_token_url,
-            refresh_token=refresh_token,
-            enc_key=enc_key,
-            redirect_uri=redirect_uri,
-            token_retrieval_url='https://login.microsoftonline.com/{tenant_id}/oauth2/token',
-            grant_type=AUTHORIZATION_CODE,  # disable-secrets-detection
-            app_name=APP_NAME,
-            base_url=base_url,
-            verify=verify,
-            proxy=proxy,
-            resource='https://management.core.windows.net',
-            scope='',
+class AzureSentinelClient(MicrosoftClient):
+    def __init__(self, server_url: str, tenant_id: str, client_id: str,
+                 client_secret: str, subscription_id: str,
+                 resource_group_name: str, workspace_name: str,
+                 verify: bool = True, proxy: bool = False):
+
+        if not server_url:
+            server_url = f'https://management.azure.com/subscriptions/{subscription_id}/'\
+                         f'resourceGroups/{resource_group_name}/providers/Microsoft.OperationalInsights/workspaces/'\
+                         f'{workspace_name}/providers/Microsoft.SecurityInsights'
+
+        super().__init__(
             tenant_id=tenant_id,
-            auth_code=auth_code,
-            ok_codes=(200, 201, 202, 204, 400, 401, 403, 404)
+            auth_id=client_id,
+            enc_key=client_secret,
+            self_deployed=True,
+            grant_type=CLIENT_CREDENTIALS,
+            token_retrieval_url=TOKEN_RETRIEVAL_URI,
+            base_url=server_url,
+            scope=Scopes.management_azure,
+            ok_codes=(200, 201, 202, 204),
+            verify=verify,
+            proxy=proxy
         )
 
-    def http_request(self, method, url_suffix=None, full_url=None, params=None, data=None, is_get_entity_cmd=False):
-        if not params:
-            params = {}
+    def http_request(self, method, url_suffix=None, full_url=None, params=None, data=None):
         if not full_url:
+            params = params or {}
```

Comment:
It does not have the same logic as the previous code, are you sure that's what you meant?

Reply:
Yes 
the params is only needed in case of not full_url